src/app/menu/menu.component.html

```diff
@@ -17,12 +17,12 @@
         </div>
         <div class="row">
             <div class="menu-button">
-                <img class="hex" src="../../assets/img/malaria-101.png" (click)="navigateToPage('/malaria-101')">
-                <button class="but malaria-btn" (click)="navigateToPage('/malaria-101')">
+                <img class="hex" [src]="calculateButtonBackground(2, 'malaria-101')" (click)="navigateToPage('/malaria-101')">
+                <button [ngClass]="calculateButtonClasses(2, 'malaria-btn')" (click)="navigateToPage('/malaria-101')" [disabled]="!checkStageStatus(2)">
                     {{language?.malaria101.heading}}
                 </button>
-                <div class="outer-circle malaria-outer-circle">
-                    <div class="inner-circle malaria-inner-circle">
+                <div [ngClass]="calculateOuterClasses(2, 'malaria-outer-circle')">
+                    <div [ngClass]="calculateInnerClasses(2, 'malaria-inner-circle')">
                         <div *ngFor="let level of createRange(3); let idx=index" class="arc level{{idx+1}}" [ngClass]="stage>2 || (stage===2 && activity>=idx+1) ? 'completed' : ''"></div>
```

Comment:
please create a method for this
(stage===3 && activity>=idx+1) ? 'completed' : ''"
make it return a boolean

this can be reused for next method as well.. In general don't keep complicated logics in the Html part.